lib/Transforms/MemoryBanking.cpp

```diff
@@ -426,45 +494,48 @@ struct BankAffineLoadPattern
     auto loadIndices = loadOp.getIndices();
     MemRefType originalMemRefType = loadOp.getMemRefType();
     int64_t memrefRank = originalMemRefType.getRank();
-    ArrayRef<int64_t> shape = originalMemRefType.getShape();
 
-    auto bankingDimension =
-        getSpecifiedOrDefaultBankingDim(bankingDimensionOpt, memrefRank, shape);
+    BankingConfigAttributes currBankingConfig =
+        getMemRefBankingConfig(originalMem);
+    if (!currBankingConfig.factors) {
+      assert(!currBankingConfig.dimensions &&
+             "A banking dimension must be paired with a factor");
```

Comment:
This was dicey to me. Why aren't these passed as pairs already? Is it really easier to have these as two lists for the compiler / frontend user?

Reply:
sometimes users might only pass in `factors` without `dimensions`;
and it also adheres to the existing HLS tool by Vitis: https://docs.amd.com/r/en-US/ug1399-vitis-hls/Array-Partitioning

Reply:
Ok so to check my understanding, `factors=4` would apply it to every dimension? Regardless it'd be nice if this was documented

Reply:
Just refactored the logic and ready for another round of review!